Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/WalletEthereumProviderScript.js

```diff
@@ -89,13 +126,52 @@ if (window.isSecureContext) {
+    };
+    let provider = window.ethereum
+    const event = new CustomEvent("eip6963:announceProvider", {
+      detail: $Object.freeze({ info, provider }),
+    });
+    window.dispatchEvent(event);
```

Comment:
It's less likely a page would do this and more an extension wallet (for desktop). In this case, we're the only ones who can inject scripts into this page other than some sort of malicious XSS on the page. It would be good to mitigate that with an addition of object.freeze to the event itself too.

Reply:
Updated to freeze the event itself in https://github.com/brave/brave-ios/pull/8034/commits/32254b5ead2b068a57c7d347f340767c0fdb6bfd

Reply:
I think we need to freeze `window.dispatchEvent` instead, or rather create a copy of `window.dispatchEvent` freeze it and call the copy instead of `window.dispatchEvent`. Similar to how we call `postMessage()` in webkit message handlers. Because this code, executed by another wallet, can overwrite `window.dispatchEvent` itself and prevent events from dispatching/modify event objects:
```javascript
window.dispatchEvent = function(arguments) {
    console.log(arguments)
    var ret = oldDispatch(arguments)
    return ret
}
```

Reply:
Updated to use a frozen copy of `dispatchEvent` and `addEventListener` in [1b21e22](https://github.com/brave/brave-ios/pull/8034/commits/1b21e22173c2d3ce85650473aa6d4eaf3635aee9). cc @Brandon-T for `__firefox__.js` changes.

Reply:
![image](https://media.giphy.com/media/3orif8wvSkpJjUScvu/giphy.gif)